api/src/main/java/org/openmrs/module/mksreports/reports/PatientHistoryReportManager.java

```diff
@@ -0,0 +1,236 @@
+	
+	public final static String ENCOUNTER_UUID_LABEL = "encounter_uuid";
+	
+	public final static String ENCOUNTER_PROVIDER_LABEL = "provider_name";
+	
+	public final static String ENCOUNTER_TYPE_UUID_LABEL = "encounter_type_uuid";
+	
+	public final static String ENCOUNTERTYPE_NAME_LABEL = "encounter_type_name";
+	
+	public final static String ENCOUNTER_DATETIME_LABEL = "encounter_datetime";
+	
+	public final static String OBS_VALUE_LABEL = "obs_value";
+	
+	public final static String OBS_DATETIME_LABEL = "obs_datetime";
+	
+	public final static String OBS_DATATYPE_LABEL = "concept_datatype";
+	
+	public final static String OBS_NAME_LABEL = "concept_name";
+	
+	public final static String OBS_PROVIDER_LABEL = "provider_name";
+	
+	@Autowired
+	private EncounterDataLibrary encounterDataLibrary;
+	
+	@Autowired
+	private BuiltInPatientDataLibrary builtInPatientDataLibrary;
+	
+	@Autowired
+	private ObsDataLibrary obsDataLibrary;
+	
+	@Autowired
+	BasePatientDataLibrary basePatientDataLibrary;
+	
+	@Override
+	public boolean isActive() {
+		return super.isActive();
+	}
+	
+	@Override
+	public String getVersion() {
+		return "1.0.0-SNAPSHOT";
+	}
+	
+	@Override
+	public String getUuid() {
+		return "ae2152c1-1a91-4dc9-96d2-dfc38820648e";
+	}
+	
+	@Override
+	public String getName() {
+		return "HC1 Patient History";
+	}
+	
+	@Override
+	public String getDescription() {
+		return StringUtils.EMPTY;
+	}
+	
+	private Parameter getLocationParameter() {
+		return new Parameter("locationList", "Visit Location", Location.class, List.class, null);
+	}
+	
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> params = new ArrayList<Parameter>();
+		params.add(getLocationParameter());
+		return params;
+	}
+	
+	@Override
+	public ReportDefinition constructReportDefinition() {
+		ReportDefinition reportDef = new ReportDefinition();
+		reportDef.setUuid(this.getUuid());
+		reportDef.setName(REPORT_DEFINITION_NAME);
+		reportDef.setDescription(this.getDescription());
+		
+		Map<String, Object> mappings = new HashMap<String, Object>();
+		MessageSourceService i18nTranslator = Context.getMessageSourceService();
+		
+		// Create dataset definitions
+		PatientDataSetDefinition patientDataSetDef = createDemographicsDataSetDefinition(i18nTranslator);
+		PatientHistoryObsAndEncounterDataSetDefinition obsDataSetDef = createObsAndEncounterDataSetDefinition();
+		PatientHistoryEncounterAndVisitDataSetDefinition encountersDatasetSetDef = createEncounterAndVisitDataSetDefinition();
+		
+		// Add datasets to the report
+		reportDef.addDataSetDefinition(DATASET_KEY_DEMOGRAPHICS, patientDataSetDef, mappings);
+		reportDef.addDataSetDefinition(DATASET_KEY_OBS, obsDataSetDef, mappings);
+		reportDef.addDataSetDefinition(DATASET_KEY_ENCOUNTERS, encountersDatasetSetDef, new HashMap<String, Object>());
+		
+		return reportDef;
+	}
+	
+	private PatientHistoryEncounterAndVisitDataSetDefinition createEncounterAndVisitDataSetDefinition() {
```

Comment:
We need to challenge whether this class was needed: `PatientHistoryEncounterAndVisitDataSetDefinition`, and find out whether we can use `EncounterDataSetDefinition` directly.

Reply:
@mks-d , I removed these two customs datasetDefinition and replaced by the correct ones